hasher.go

```diff
@@ -415,3 +483,88 @@ func (h *Hasher) merkleizeImpl(dst []byte, input []byte, limit uint64) []byte {
 	}
 	return append(dst, res...)
 }
+
+func merkleizeInput(input []byte, limit uint64) []byte {
+	elemCount := len(input) / 32
```

Comment:
Don't you have an off-by one here? if `len(input) = 1` don't you want to have one chunk? I would have expected here something like `(len(input) + 31)/32`